core/src/SYCL/Kokkos_SYCL_ParallelFor_MDRange.hpp

```diff
@@ -153,12 +153,29 @@ class Kokkos::Impl::ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>,
             {global_x, global_y, global_z}, {local_x, local_y, local_z})
             .exec_range();
       });
-    });
+    };
+
+    if constexpr (Policy::is_graph_kernel::value) {
+      sycl::ext::oneapi::experimental::command_graph<
+          sycl::ext::oneapi::experimental::graph_state::modifiable>& graph =
+          Impl::get_sycl_graph_from_kernel(*this);
+      std::optional<sycl::ext::oneapi::experimental::node>& graph_node =
+          Impl::get_sycl_graph_node_from_kernel(*this);
+      KOKKOS_ENSURES(!graph_node);
+      graph_node = graph.add(cgh_lambda);
+      KOKKOS_ENSURES(graph_node);
+      // FIXME_SYCL_GRAPH not yet implemented in the compiler
+      //   KOKKOS_ENSURES(graph_node.get_type() ==
+      //   sycl::ext::oneapi::experimental::node_type::kernel)
+      return {};
```

Comment:
I left it here since a proper reduction or scan implementation needs to create multiple nodes anyway. It might make sense to put it somewhere else and refactor it once we implement those types of kernels properly.